### PR TITLE
mcs: disable PD scheduling when enabling scheduling service

### LINKED ISSUE #7041: tests/integrations/mcs/scheduling is unstable
### Which jobs are failing
```
2023-09-04T10:01:53.5628693Z [2023/09/04 10:01:12.712 +00:00] [FATAL] [log.go:87] [panic] [recover="\"invalid memory address or nil pointer dereference\""] [stack="github.com/tikv/pd/pkg/utils/logutil.LogPanic\n\t/home/runner/work/pd/pd/pkg/utils/logutil/log.go:87\nruntime.gopanic\n\t/opt/hostedtoolcache/go/1.21.0/x64/src/runtime/panic.go:920\nruntime.panicmem\n\t/opt/hostedtoolcache/go/1.21.0/x64/src/runtime/panic.go:261\nruntime.sigpanic\n\t/opt/hostedtoolcache/go/1.21.0/x64/src/runtime/signal_unix.go:861\ngithub.com/tikv/pd/pkg/mcs/scheduling/server.(*Server).updateAPIServerMemberLoop\n\t/home/runner/work/pd/pd/pkg/mcs/scheduling/server/server.go:166"]
```

### CI link
https://github.com/tikv/pd/actions/runs/6071828360/job/16470650372?pr=7036

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: ref #5839, close #7041.

### What is changed and how does it work?

This PR disables the PD scheduling when enabling the scheduling service. It also registers the schedulers for scheduling service.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/statistics/store_collection.go [hunk 1]
```diff
@@ -147,8 +147,11 @@ func (s *storeStatistics) Observe(store *core.StoreInfo, stats *StoresStats) {
 		storeStatusGauge.WithLabelValues(storeAddress, id, "store_slow_trend_result_rate").Set(slowTrend.ResultRate)
 	}
 
+	if len(stats) == 0 {
```
Comment (lhy1024) [inline pkg/statistics/store_collection.go:150]: how about observing store flow with another function?

File: server/cluster/cluster.go [hunk 2]
```diff
@@ -590,8 +596,10 @@ func (c *RaftCluster) LoadClusterInfo() (*RaftCluster, error) {
 	)
 	for _, store := range c.GetStores() {
 		storeID := store.GetID()
```
Comment (lhy1024) [inline server/cluster/cluster.go:598]: also put in `if !c.isAPIServiceMode {`?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

**Review by lhy1024 — APPROVED** [review]

**Review by lhy1024 — COMMENTED** [review]
TestSchedulerConfigWatch is failed

**Comment by rleungx** [comment]
@JmPotato PTAL

File: tests/integrations/mcs/scheduling/config_test.go [hunk 3]
```diff
@@ -137,17 +137,10 @@ func (suite *configTestSuite) TestSchedulerConfigWatch() {
 	re.NoError(err)
 	// Get all default scheduler names.
 	var (
-		schedulerNames      []string
-		schedulerController = suite.pdLeaderServer.GetRaftCluster().GetCoordinator().GetSchedulersController()
+		schedulerNames, _, _ = suite.pdLeaderServer.GetRaftCluster().GetStorage().LoadAllScheduleConfig()
```
Comment (JmPotato) [inline tests/integrations/mcs/scheduling/config_test.go:140]:
```suggestion
		var schedulerNames, _, _ = suite.pdLeaderServer.GetRaftCluster().GetStorage().LoadAllScheduleConfig()
```

File: server/cluster/cluster.go [hunk 4]
```diff
@@ -1064,22 +1090,24 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
...
 	}
-	c.coordinator.GetSchedulersController().CheckTransferWitnessLeader(region)
 
 	hasRegionStats := c.regionStats != nil
 	// Save to storage if meta is updated, except for flashback.
 	// Save to cache if meta or leader is updated, or contains any down/pending peer.
 	// Mark isNew if the region in cache does not have leader.
 	isNew, saveKV, saveCache, needSync := regionGuide(region, origin)
-	if !saveKV && !saveCache && !isNew {
+	if !c.isAPIServiceMode && !saveKV && !saveCache && !isNew {
```
Comment (JmPotato) [inline server/cluster/cluster.go:1110]: Shouldn't it be like this?

```suggestion
	if c.isAPIServiceMode || (!saveKV && !saveCache && !isNew) {
```

Comment (rleungx) [reply server/cluster/cluster.go:1110]: If it is API mode, we should skip it.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**